cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java

```diff
@@ -28,9 +28,13 @@
 import io.ballerina.projects.ProjectException;
 import io.ballerina.projects.ProjectKind;
 import io.ballerina.projects.directory.SingleFileProject;
+import io.ballerina.projects.internal.DefaultDiagnosticResult;
```

Comment:
Is this intentional?  I.e.depending on internal packages.

Reply:
Yes, but better we avoid it. Refactored the implementation to use the public APIs.